anndata/_core/anndata.py

```diff
@@ -854,7 +865,24 @@ def var_names(self, names: Sequence[str]):
     @property
     def uns(self) -> MutableMapping:
         """Unstructured annotation (ordered dictionary)."""
-        return self._uns
+        uns = self._uns
+        # Special case for old behaviour of neighbors
+        if "neighbors" in uns:
```

Comment:
So this is created every time when` .uns` is accessed. Is there any way to create it once and then just return on every access?

Reply:
Two reasons for this:

There's a reference to the anndata object. This makes sure it's pointing to the right one. If it was always the same we could have trouble with statements like:

```python
a: AnnData

b = a[:100].copy()
b.uns["neighbors"]["connectivities"]  # This should get b.obsp["connectivities"]
```

Second is that copy and deep copy are already weird for DictViews. This simplifies it a bit.

Reply:
Ok, thats reasonable.